synfig-core/src/synfig/distance.cpp

```diff
@@ -111,11 +111,15 @@ Distance::operator=(const synfig::String& str)
 }
 
 synfig::String
-Distance::get_string(int digits)const
+Distance::get_string(int digits, bool remove_trailing_zeroes_, bool force_decimal_point, bool has_unit)const
 {
 	digits=synfig::clamp(digits,0,9);
 	String fmt(strprintf("%%.%01df",digits));
 	String str(strprintf(fmt.c_str(),value_));
+	if(remove_trailing_zeroes_){
+		return remove_trailing_zeroes(strprintf("%s%s",str.c_str(),system_name(system_).c_str()),
```

Comment:
You should remove trailing zeroes **before** adding `system_name(...)`

Reply:
oh, I get what you mean now. yes makes sense.

Reply:
done